File: backend/app/schemas/document_request.py
"""Document request Pydantic schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class DocumentRequestCreate(BaseModel):
    description: str = Field(..., min_length=1, max_length=2000)


class DocumentRequestSubmit(BaseModel):
    evidence_ids: list[UUID] = Field(..., min_length=1)


class DocumentRequestOut(BaseModel):
    id: UUID
    tenant_id: UUID
    claim_id: UUID
    requested_by_user_id: UUID
    description: str
    status: str
    resolved_at: datetime | None = None
    created_at: datetime
    updated_at: datetime | None = None

    model_config = {"from_attributes": True}


class DocumentRequestListResponse(BaseModel):
    items: list[DocumentRequestOut]
    total: int
    page: int
    limit: int

File: frontend/types/document-request.ts
export interface DocumentRequest {
  id: string
  tenant_id: string
  claim_id: string
  requested_by_user_id: string
  description: string
  status: 'pending' | 'submitted' | 'waived'
  resolved_at: string | null
  created_at: string
  updated_at: string | null
}

export interface DocumentRequestListResponse {
  items: DocumentRequest[]
  total: number
  page: number
  limit: number
}

export interface DocumentRequestCreatePayload {
  description: string
}
